File: COBWEB-3D/Module1.vb
Module Module1
    Sub asrproduce(ByVal agent As Integer)
        If Form1.total < generator.maxcell Then
            Form1.total = Form1.total + 1

            Dim x As Integer = CInt(Math.Floor((Form1.xn) * Rnd())) + 1
            Dim y As Integer = CInt(Math.Floor((Form1.yn) * Rnd())) + 1
            Dim z As Integer = CInt(Math.Floor((Form1.zn) * Rnd())) + 1

            Dim dx As Integer
            Dim dy As Integer
            Dim dz As Integer
            'Dim a As Integer = generator.action(ag1, ag2, 2)
            'look here


            Dim rangexupper As Integer = generator.agentrange(agent, 0, 1)
            Dim rangexlower As Integer = generator.agentrange(agent, 0, 0)
            Dim rangeyupper As Integer = generator.agentrange(agent, 1, 1)
            Dim rangeylower As Integer = generator.agentrange(agent, 1, 0)
            Dim rangezupper As Integer = generator.agentrange(agent, 2, 1)
            Dim rangezlower As Integer = generator.agentrange(agent, 2, 0)

            dx = CInt(Math.Floor((rangexupper - rangexlower + 1) * Rnd())) + rangexlower
            dy = CInt(Math.Floor((rangeyupper - rangeylower + 1) * Rnd())) + rangeylower
            dz = CInt(Math.Floor((rangezupper - rangezlower + 1) * Rnd())) + rangezlower



            Dim number As Integer = 0
            Do While generator.occupied(x, y, z) = True And number < generator.maxcell
                number = number + 1
                x = CInt(Math.Floor((Form1.xn) * Rnd())) + 1
                y = CInt(Math.Floor((Form1.yn) * Rnd())) + 1
                z = CInt(Math.Floor((Form1.zn) * Rnd())) + 1
            Loop


            generator.occupied(x, y, z) = True

            Dim d As Integer = CInt(Math.Floor((6) * Rnd())) + 1
            generator.agentlocation(Form1.total, 0) = x
            generator.agentlocation(Form1.total, 1) = y
            generator.agentlocation(Form1.total, 2) = z
            generator.agentlocation(Form1.total, 3) = d
            generator.agentlocation(Form1.total, 4) = agent
            generator.agentlocation(Form1.total, 5) = dx
            generator.agentlocation(Form1.total, 6) = dy
            generator.agentlocation(Form1.total, 7) = dz
            generator.agentlocation(Form1.total, 8) = generator.initialenergy(agent)
            generator.agentlocation(Form1.total, 9) = 0
            generator.agentlocation(Form1.total, 10) = 0
        End If
    End Sub
End Module

File: COBWEB 3D/CS/Module1.cs
using System.Collections.Generic;
using System;
using System.Linq;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using System.Xml.Linq;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;

namespace COBWEB_3D
{
	sealed class Module1
	{
		static public void asrproduce(int agent)
		{
			if (Form1.Default.total < generator.Default.maxcell)
			{
				Form1.Default.total++;
				
				int x = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((Form1.Default.xn) * VBMath.Rnd())))) + 1;
				int y = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((Form1.Default.yn) * VBMath.Rnd())))) + 1;
				int z = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((Form1.Default.zn) * VBMath.Rnd())))) + 1;
				
				int dx;
				int dy;
				int dz;
				//Dim a As Integer = generator.action(ag1, ag2, 2)
				//look here
				
				
				int rangexupper = generator.Default.agentrange[agent, 0, 1];
				int rangexlower = generator.Default.agentrange[agent, 0, 0];
				int rangeyupper = generator.Default.agentrange[agent, 1, 1];
				int rangeylower = generator.Default.agentrange[agent, 1, 0];
				int rangezupper = generator.Default.agentrange[agent, 2, 1];
				int rangezlower = generator.Default.agentrange[agent, 2, 0];
				
				dx = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((rangexupper - rangexlower + 1) * VBMath.Rnd())))) + rangexlower;
				dy = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((rangeyupper - rangeylower + 1) * VBMath.Rnd())))) + rangeylower;
				dz = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((rangezupper - rangezlower + 1) * VBMath.Rnd())))) + rangezlower;
				
				
				
				int number = 0;
				while (generator.Default.occupied[x, y, z] == true && number < generator.Default.maxcell)
				{
					number++;
					x = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((Form1.Default.xn) * VBMath.Rnd())))) + 1;
					y = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((Form1.Default.yn) * VBMath.Rnd())))) + 1;
					z = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((Form1.Default.zn) * VBMath.Rnd())))) + 1;
				}
				
				
				generator.Default.occupied[x, y, z] = true;
				
				int d = (System.Convert.ToInt32(Math.Floor(System.Convert.ToDecimal((6) * VBMath.Rnd())))) + 1;
				generator.Default.agentlocation[Form1.Default.total, 0] = x;
				generator.Default.agentlocation[Form1.Default.total, 1] = y;
				generator.Default.agentlocation[Form1.Default.total, 2] = z;
				generator.Default.agentlocation[Form1.Default.total, 3] = d;
				generator.Default.agentlocation[Form1.Default.total, 4] = agent;
				generator.Default.agentlocation[Form1.Default.total, 5] = dx;
				generator.Default.agentlocation[Form1.Default.total, 6] = dy;
				generator.Default.agentlocation[Form1.Default.total, 7] = dz;
				generator.Default.agentlocation[Form1.Default.total, 8] = generator.Default.initialenergy[agent];
				generator.Default.agentlocation[Form1.Default.total, 9] = 0;
				generator.Default.agentlocation[Form1.Default.total, 10] = 0;
			}
		}
	}
	
}
